crates/iota/src/name_commands.rs

```diff
@@ -259,22 +304,49 @@ impl NameCommand {
                 )
             }
             Self::Auction(AuctionCommand::Metadata { domain }) => {
-                NameCommandResult::AuctionMetadata(get_auction(&domain, context).await?)
+                let graphql_client = SimpleClient::new(
+                    context
+                        .active_env()?
+                        .graphql()
+                        .as_ref()
+                        .ok_or_else(|| anyhow::anyhow!("missing graphql url in IotaEnv"))?,
+                );
+                NameCommandResult::AuctionMetadata(
+                    get_auction_house(&iota_client, &graphql_client)
+                        .await?
+                        .get_auction(&domain, &iota_client)
+                        .await?,
+                )
             }
             Self::Auction(AuctionCommand::Start {
                 domain,
                 amount,
                 coin,
                 opts,
             }) => {
-                let auction_package_address = get_auction_package_address(context).await?;
+                let graphql_client = SimpleClient::new(
+                    context
+                        .active_env()?
+                        .graphql()
+                        .as_ref()
+                        .ok_or_else(|| anyhow::anyhow!("missing graphql url in IotaEnv"))?,
+                );
```

Comment:
This repeated code is really annoying, does really make a lot of sense to extract it since it's used in only one place atm